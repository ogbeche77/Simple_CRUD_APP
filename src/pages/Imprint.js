import React from "react";

function Imprint() {
  return <div>&copy; {new Date().getFullYear()} Copyright</div>;
}

export default Imprint;
