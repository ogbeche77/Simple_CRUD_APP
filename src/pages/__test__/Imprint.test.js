import Imprint from "../Imprint";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";

test("should render imprint component", () => {
  const sampleTest = "I am a sampleTester";

  const div = document.createElement("div");
  render(<Imprint sampleTest={sampleTest} />, div);
  const imprintComponent = screen.getByTestId("sample-item");
  expect(imprintComponent).toBeInTheDocument();
  expect(imprintComponent).toHaveTextContent("I am a sampleTester");
  expect(imprintComponent).toContainHTML("div");
});

it("matches imprint component snapshot", () => {
  const sampleTest = "I am a sampleTester";
  const imprintSnapshot = TestRenderer.create(
    <Imprint sampleTest={sampleTest}></Imprint>
  ).toJSON();
  expect(imprintSnapshot).toMatchSnapshot();
});
