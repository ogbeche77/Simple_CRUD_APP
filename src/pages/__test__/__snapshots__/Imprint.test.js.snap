// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches imprint component snapshot 1`] = `
Array [
  " ",
  <div>
    Â© 
    2021
     Copyright
  </div>,
  <div
    data-testid="sample-item"
  >
    I am a sampleTester
  </div>,
]
`;
